### PR TITLE
Ignore errors for hover

### PR DESCRIPTION
**Pull request by grouma**
Closes https://github.com/flutter/devtools/issues/2796

### CODE REVIEW

File: packages/devtools_app/lib/src/debugger/codeview.dart [hunk 1]
```diff
@@ -610,6 +610,7 @@ class _LineItemState extends State<LineItem> {
       if (word != '') {
         try {
           final response = await _debuggerController.evalAtCurrentFrame(word);
+          if (response.json['kind'] == 'error') return;
```
Comment (jacob314) [inline packages/devtools_app/lib/src/debugger/codeview.dart:613]: I would have thought you could use the type of the response.

Comment (grouma) [reply packages/devtools_app/lib/src/debugger/codeview.dart:613]: Yeah that's what I thought too but the type is `Response`.

Comment (grouma) [reply packages/devtools_app/lib/src/debugger/codeview.dart]: Found a better alternative.

File: packages/devtools_app/lib/src/debugger/codeview.dart [hunk 2]
```diff
@@ -610,6 +610,7 @@ class _LineItemState extends State<LineItem> {
       if (word != '') {
         try {
           final response = await _debuggerController.evalAtCurrentFrame(word);
+          if (response is ErrorRef) return;
```
Comment (jacob314) [inline packages/devtools_app/lib/src/debugger/codeview.dart:613]: not that it will happen, but it should be `is! InstanceRef` for robustness.

**Review by jacob314 — APPROVED** [review]
lgtm

**Comment by grouma** [comment]
Merging on red as this should not impact tests and we want to roll this internally.

**Pull request merged**